props/blaster.h

```diff
@@ -260,9 +233,65 @@ class Blaster : public PROP_INHERIT_PREFIX PropBase {
     }
   }
 
+   RefPtr<BufferedWavPlayer> auto_player_;
+
+  uint32_t auto_time_;
+  virtual void Fire() {
+    CheckEmpty();
+    if (empty_) return;
+#ifdef ENABLE_MOTION
+#ifdef BLASTER_JAM_PERCENTAGE
+    // If we're already jammed then we don't need to recheck. If we're not jammed then check if we just jammed.
+    is_jammed_ = is_jammed_ ? true : CheckJam(BLASTER_JAM_PERCENTAGE);
+    if (is_jammed_) {
+      SaberBase::DoEffect(EFFECT_JAM, 0);
+      return;
+    }
+#endif
+#endif
+    if (blaster_mode == MODE_AUTO) {
+      // Set up the autofire pairing if the font suits it.
+      SelectAutoFirePair(); 
+      SaberBase::SetLockup(LOCKUP_AUTOFIRE);
+      SaberBase::DoBeginLockup();
+      auto_firing_ = true;
+      auto_time_ = millis();
+
+      // GET auto.wav here
+      if (!auto_player_) {
+        auto_player_ = GetWavPlayerPlaying(&SFX_auto);
+        }
+    } else {
+      if (blaster_mode == MODE_STUN) {
+        SaberBase::DoEffect(EFFECT_STUN, 0);
+        STDOUT << "******** STUN - Remaining shots = " << GetBulletCount() << "\n";
+      } else {
+        SaberBase::DoEffect(EFFECT_FIRE, 0);
+        STDOUT << "******** FIRE - Remaining shots = " << GetBulletCount() << "\n";
+      }
+      shots_fired_++;
+    }
+  }
+
   void Loop() override {
     PropBase::Loop();
     PollNextAction();
+    if (auto_firing_) {
+      if (&auto_player_) {
```

Comment:
This takes the address of "auto_player_", which is always true.
Just do if(auto_player_)

Reply:
But that's not in the scope. That's what i kept running up against. Not sure how to do that, spent hours :(

Reply:
Why do you think it's not in scope?
Do you get a compiler error?
What is that error?

Reply:
> Why do you think it's not in scope? Do you get a compiler error? What is that error?

Well, of course it's working fine now! 
Man, over and over I couldn't get it to work before. No error, it just never behaved as if the if clause was true.
I put an `else` on that `if (auto_player_)` to print a message " NOPE! no auto_player_"
And that is all I would get, the counting bullets and checking empty part wasn't happening.
Must have learned or done something right overnight I guess. Like moving the declaration ouf of the Fire() function out to global scope. Anyway, MOVING ON. 
Thank you.